docs/source/advanced.rst

```diff
@@ -258,23 +258,217 @@ of :py:class:`scikeras.wrappers.BaseWrappers`. Currently, they are:
+   `sklearn.model_selection.RandomizedSearchCV`; `SGD(learning_rate=0.5)` can
+   not be used with that class.
+
+
+Losses
+++++++
+
+.. code:: python
+
+    from tensorflow.keras.losses import BinaryCrossentropy, CategoricalCrossentropy
+
+    clf = KerasClassifier(
+        ...
+        loss=BinaryCrossentropy,
+        loss__label_smoothing=0.1,  # results in BinaryCrossentropy(label_smoothing=0.1)
+    )
+    # pure Keras object only syntax is still supported, but parameters can't be tuned
+    clf = KerasClassifier(
+        ...
+        loss=BinaryCrossentropy(label_smoothing=0.1),
+    )
+
+Keras, and SciKeras by extension, support passing a single loss (which will be applied to all outputs)
+or a loss for each output by passing a list of losses or a dict of losses.
+
+Additionally, SciKeras supports routed parameters to each individual loss, or to all losses together.
+
+.. code:: python
+
+    from tensorflow.keras.losses import BinaryCrossentropy, CategoricalCrossentropy
+
+    clf = KerasClassifier(
+        ...
         loss=[BinaryCrossentropy, CategoricalCrossentropy],
         loss__from_logits=True,  # BinaryCrossentropy(from_logits=True) & CategoricalCrossentropy(from_logits=True)
-        loss__label_smoothing=0.1,  # passed to each sub-item, i.e. `loss=[l(label_smoothing=0.1) for l in loss]`
         loss__1__label_smoothing=0.5,  # overrides the above, results in CategoricalCrossentropy(label_smoothing=0.5)
     )
+    # or
+    clf = KerasClassifier(
+        ...
+        loss={"out1": BinaryCrossentropy, "out2": CategoricalCrossentropy},
+        loss__from_logits=True,  # BinaryCrossentropy(from_logits=True) & CategoricalCrossentropy(from_logits=True)
+        loss__out2__label_smoothing=0.5,  # overrides the above, results in CategoricalCrossentropy(label_smoothing=0.5)
+    )
+
+With this parameter routing in place, you can now use hyperparameter tuning on each loss' parameters.
 
+Metrics
++++++++
+
+Metrics have similar semantics to losses, but multiple metrics per output are supported.
+Here are several support use cases:
+
+.. code:: python
+
+    from tensorflow.keras.metrics import BinaryAccuracy, AUC
+
+    clf = KerasClassifier(
+        ...
+        metrics=BinaryAccuracy,
+        metrics__threashold=0.65,
+    )
+    # or to apply multiple metrics to all outputs
+    clf = KerasClassifier(
+        ...
+        metrics=[BinaryAccuracy, AUC],  # both applied to all outputs
+        metrics__0__threashold=0.65,
+    )
+    # or for different metrics to each output
+    clf = KerasClassifier(
+        ...
+        metrics={"out1": BinaryAccuracy, "out2": [BinaryAccuracy, AUC]},
+        metrics__out1__threashold=0.65,
+        metrics__out2__0__threashold=0.65,
+    )
+    # assuming you ordered your outputs like [out1, out2], this is equivalent to
+    clf = KerasClassifier(
+        ...
+        metrics=[BinaryAccuracy, [BinaryAccuracy, AUC]],
```

Comment:
```suggestion
        metrics=[BinaryAccuracy, [AUC, BinaryAccuracy]],
```

I think you should make sure AUC is selected. That way, when the reader tries to understand what object `metrics__1__0` selects, they can know if they're right or not.

Reply:
That sounds good.